### ISSUE TITLE
Daily build, insert citation: previously deleted Zotero items still show up at Zotero root

### ISSUE THREAD

**Issue by florisvdh** [rstudio/rstudio#7722]
### System details

```
RStudio Edition : Desktop
RStudio Version : 1.4.774
OS Version : Linux Mint 20 (~Ubuntu 20.04)
R Version : 4.0.2
```

### Steps to reproduce the problem

1. Use the Zotero database from #7720 and open Zotero
1. Press ctrl+shift+F8 to insert citation
1. Click on 'Zotero' in the dialog box

### Describe the problem in detail

While the Zotero database effectively has only 21 items, and an empty trash, old items still show up that had been deleted previously, resulting in a list of way more than 21 items (in the isnert-citation dialog).

### Describe the behavior you expected

The showed items should just be the same that can be found in the Zotero database.

- [X] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [X] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [X] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by florisvdh** [comment]
I tested this in 1.4.823, but the issue was still there. @R-Hannibal

**Comment by R-Hannibal** [comment]
Verified Not Fixed on: RStudio Version - 1.4.823.

Re-opening based on the feedback from @florisvdh (thank you for the assist!🙌). I confirmed that the code fix was in the version tested, 1.4.823.

**Comment by mikebessuille** [comment]
Charles believes this can be moved out.  Per JJ:  "It's not harmful (user can just import a reference that they have already deleted -- the import will still "work", it's just something that should be screened out). Definitely needs fixing but is not critical."   Moving to 1.5

**Comment by AndrewMcClain** [comment]
Verified fixed in 2021.09.0, no longer shows deleted Zotero entries
